tests/Common/DataFixtures/TestEntity/User.php

```diff
@@ -43,13 +51,18 @@ class User
      *
      * @var Collection<int, User>
      */
+    #[ORM\ManyToMany(targetEntity: self::class, inversedBy: 'authors')]
+    #[ORM\JoinTable(name: 'author_reader', schema: 'readers')]
```

Comment:
Is SQLite supposed to support this (The `schema` argument)?

Reply:
I reproduce the issue when using ORM 3 and DBAL 3, so it's not related to DBAL.

Reply:
Ah maybe it has to do with the `RemoveNamespacedAsset` thingy.

Reply:
ORM stopped using it in https://github.com/doctrine/orm/pull/11001

Reply:
> Is SQLite supposed to support this (The `schema` argument)?

Yes, SQLite supports schemas. You basically attach a second SQLite file and assign a schema name to it.

Reply:
Ok, but apparently, ORM 2 does not care whether it supports schemas, but whether it can emulate them: https://github.com/doctrine/orm/blob/1a4fe6e0bb67762370937a7e6cee3da40a9122d1/lib/Doctrine/ORM/Tools/SchemaTool.php#L415

In the case of SQLite, that's dynamic: https://github.com/doctrine/dbal/blob/0ac3c270590e54910715e9a1a044cc368df282b2/src/Platforms/SqlitePlatform.php#L896-L905

When running the tests on 1.7.x, it returns `false`, which I think explains why this worked before, but does not work now.

When I say it "worked", I mean it is possible to create the schema, because I soon as you try to create a user and add it as a reader of another user, it falls apart: `Unable to prepare statement: 1, no such table: readers.author_reader`.

Reply:
I tried to reproduce this in https://github.com/doctrine/orm/pull/11079, if that's fine I guess we can maybe remove the `schema` part here since doesn't seem relevant.

Reply:
I found https://github.com/doctrine/orm/pull/9259, so apparently they won't be supported in 3.0

Reply:
Schema emulation is not supported be supported in 3.0. But @derrabus says SQLite supports schemas without any emulation. And as I showed, data-fixtures is not enabling schema emulation anyway. I think the right course of action would be to determine what exactly we are trying to test here… why is there a `schema` argument in the first place?

Reply:
The git pickaxe says the test was introduced in 70c915d7a6566dfe67838ffeded8b8d654a50dec

Reply:
After much digging, I found the associated PR, which uses another commit hash, probably because of squashing: https://github.com/doctrine/data-fixtures/pull/171

Reply:
I see 2 possible ways out of this now:
a) we create seperate fixtures for separate tests.
b) we think it's cool to test the creation of the n-m table in a separate schema and attach another database as alluded to by @derrabus . That being said, I don't know if it will be that simple, because if it were, wouldn't the error message be about an unknown database? Right now the message says that there is a syntax error while creating… an index: `CREATE INDEX IDX_3155D2E5F675F31B ON readers.author_reader (author_id)`… :thinking: why is there nothing about the table?

Reply:
Created [a PR with failing tests](https://github.com/doctrine/dbal/pull/6226), might write a fix later on, unless one of you beats me to it (feel free to push to my branch if you have the necessary permissions).

Reply:
It looks really wrong. I think the real issue might be on how the ORM calls the DBAL.

Reply:
@derrabus green :)